Update xivo_install_current.sh

gesnaud opened the pull request:
Add a "-a" option (a stand for archive). Because sometimes I have uncomptabile VM infra with the ISO from mirror.xivo.io, so i had to use the  xivo_install_current.sh. But some other times, i want to choose my version (beacuse of compatibility with third party ecosystem: bluemind, xuc, novatax and so on...).

elessard1 commented:
Hi,

I'm guessing you want this to be merged in the master branch of the xivo-pbx/xivo-migration-scripts repository: you actually did a pull request on your own repository (e.g. gesnaud/xivo-migration-scripts).

Still, I've taken a look at the modification, and some things I saw:

- the variable "repo" is not defined in the case you run the script without option
- I don't like the dependency on the "bc" command: a test like "if [ "$version" \> '15.19' ]; then" should be good enough
- "let's process" should probably be replaced by "let's proceed", but in fact I would remove that echo command completely

gesnaud commented:
Hi Etienne,

Thanks to read my code, I admit that i'm not fluent in github, and i 
would'nt make a bad merge!

I make the change on point 1 & 3, but for the middle suggestion, i'm not 
totally agree:

    /if [ "$version" > '15.19' ];/


that's not working with string in our case:

    /┌─[gesnaud@gesnaud]─[~]//
    //└──> if [ "15.20" > '15.19' ]; then echo $?; else echo "false?
    "$?; fi//
    //0//
    //┌─[gesnaud@gesnaud]─[~]//
    //└──> if [ "14.28" > '15.19' ]; then echo $?; else echo "false?
    "$?; fi//
    //0/

That's why i'm using "bc", like that i can do mathmetaic operations and 
compare my float :) (you know: 13.01, 14.28,16.03 etc.)

See you,
Greg
On 06/15/16 14:06, Etienne Lessard wrote:
>
> Hi,
>
> I'm guessing you want this to be merged in the master branch of the 
> xivo-pbx/xivo-migration-scripts repository: you actually did a pull 
> request on your own repository (e.g. gesnaud/xivo-migration-scripts).
>
> Still, I've taken a look at the modification, and some things I saw:
>
>   * the variable "repo" is not defined in the case you run the script
>     without option
>   * I don't like the dependency on the "bc" command: a test like "if [
>     "$version" > '15.19' ]; then" should be good enough
>   * "let's process" should probably be replaced by "let's proceed",
>     but in fact I would remove that echo command completely
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/gesnaud/xivo-migration-scripts/pull/1#issuecomment-226167877>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/ALhgMDOkPo1PjfE8mCpgcPOH1y8xGWoVks5qL-rUgaJpZM4I2GkE>.
>

-- 
*Grégory ESNAUD*
/*Expert Télécoms*

/
310 Route de Fougères - Bâtiment Europolis
35000 Rennes
Tel: +33(0)2 30 21 00 88 <tel://+33230210088>| Mob: +33(0)6 43 47 55 90 
<tel://+33643475590>| Support: 0811859486 <tel://0811859486>

*www.avencall.com*
<http://www.avencall.com>

*Suivez nous* <https://www.facebook.com/pages/AVENCALL/180804431983700> 
<http://twitter.com/#%21/avencall> 
<http://www.linkedin.com/company/avencall>

elessard1 commented:
Note that you need to escape the > character with a backslash in the test, otherwise it doesn't give the right result:

```
# wrong result -- this actually create a file "15.19" in your current directory
$ [ "15.18" > '15.19' ]; echo $?
0

# right result
$ [ "15.18" \> '15.19' ]; echo $?
1
```

gesnaud commented:
Thanks Etienne,

You improved my bash skills :)

I update the script accordingly to the 3 points.


Thanks


On 06/15/16 14:45, Etienne Lessard wrote:
>
> Note that you need to escape the > character with a backslash in the 
> test, otherwise it doesn't give the right result:
>
> |# wrong result -- this actually create a file "15.19" in your current 
> directory $ [ "15.18" > '15.19' ]; echo $? 0 # right result $ [ 
> "15.18" \> '15.19' ]; echo $? 1 |
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/gesnaud/xivo-migration-scripts/pull/1#issuecomment-226176116>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/ALhgMOc44MglkPmvnbLPDbK1Pl47R7aGks5qL_PmgaJpZM4I2GkE>.
>

-- 
*Grégory ESNAUD*
/*Expert Télécoms*

/
310 Route de Fougères - Bâtiment Europolis
35000 Rennes
Tel: +33(0)2 30 21 00 88 <tel://+33230210088>| Mob: +33(0)6 43 47 55 90 
<tel://+33643475590>| Support: 0811859486 <tel://0811859486>

*www.avencall.com*
<http://www.avencall.com>

*Suivez nous* <https://www.facebook.com/pages/AVENCALL/180804431983700> 
<http://twitter.com/#%21/avencall> 
<http://www.linkedin.com/company/avencall>

elessard1 commented:
While preparing for merging your changes into the upstream (e.g. xivo-pbx) repository, I've found that:

* installing an archive version older than 14.18 doesn't work properly, because xivo-dist is not available on these versions, but the script try to install and use xivo-dist unconditionally. The script should either be fixed to handle this case, or it should not allow to install archive version earlier than 14.18.
* the script also calls debconf-set-selections with the preseed file from jessie: currently. this is "ok" since the files are actually the same on wheezy and jessie, but this is a bit brittle

There's also a few style issues (like the added whitespace before the "usage" function, or the added empty lines, or the use of some bashism which we try to avoid when there's a similar sh-compatible alternative), but it's fine to leave them as is, I'll "fix" them before merging.

gesnaud commented:
Hi Etienne,


 > The script should either be fixed to handle this case

that would be the cleanest way... but how specify xivo verison w/o 
xivo-dist?

 > the script also calls debconf-set-selections with the preseed file 
from jessie

Thanks, I missed that! I'll submit a change.
Note: that would'nt be nicer to have symlinks like 'stable' and 
'oldstable' in mirror.xivo.io/d-i ? It would simply this script's 
maintenance?

 > There's also a few style issues (like the added whitespace before the 
"usage" function, or the added empty lines, or the use of some bashism 
which we try to avoid when there's a similar sh-compatible alternative), 
but it's fine to leave them as is, I'll "fix" them before merging.

I fixed the whitespace before usage
I don't get the empty lines and bashism? I really like to improve 
myself, don't hesitate to show me :)


Thanks Etienne



On 06/17/16 16:02, Etienne Lessard wrote:
>
> While preparing for merging your changes into the upstream (e.g. 
> xivo-pbx) repository, I've found that:
>
>   * installing an archive version older than 14.18 doesn't work
>     properly, because xivo-dist is not available on these versions,
>     but the script try to install and use xivo-dist unconditionally.
>     The script should either be fixed to handle this case, or it
>     should not allow to install archive version earlier than 14.18.
>   * the script also calls debconf-set-selections with the preseed file
>     from jessie: currently. this is "ok" since the files are actually
>     the same on wheezy and jessie, but this is a bit brittle
>
> There's also a few style issues (like the added whitespace before the 
> "usage" function, or the added empty lines, or the use of some bashism 
> which we try to avoid when there's a similar sh-compatible 
> alternative), but it's fine to leave them as is, I'll "fix" them 
> before merging.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/gesnaud/xivo-migration-scripts/pull/1#issuecomment-226776805>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/ALhgMAmHf9kcd8speOTgOrsRyHpxLBjgks5qMqjjgaJpZM4I2GkE>.
>

-- 
*Grégory ESNAUD*
/*Expert Télécoms*

/
310 Route de Fougères - Bâtiment Europolis
35000 Rennes
Tel: +33(0)2 30 21 00 88 <tel://+33230210088>| Mob: +33(0)6 43 47 55 90 
<tel://+33643475590>| Support: 0811859486 <tel://0811859486>

*www.avencall.com*
<http://www.avencall.com>

*Suivez nous* <https://www.facebook.com/pages/AVENCALL/180804431983700> 
<http://twitter.com/#%21/avencall> 
<http://www.linkedin.com/company/avencall>

elessard1 commented on xivo_install_current.sh:
```diff
@@ -63,28 +81,29 @@ install_xivo () {
 }
 
-usage() {
+  usage() {
```
When I was talking about "extra whitespace" before the usage funtion, I was talking about this line.

elessard1 commented on xivo_install_current.sh:
```diff
@@ -4,31 +4,49 @@ mirror_xivo="http://mirror.xivo.io"
 }
 
+
+
```
Here, extra newlines have been added. They probably should be removed, since functions seem to  only be separated by 1 newline in the other part of the file.

elessard1 commented on xivo_install_current.sh:
```diff
@@ -4,31 +4,49 @@ mirror_xivo="http://mirror.xivo.io"
     fi
-    if [ $version != '8' ]; then
+
+    if [[ $xivo_target_version == 'dev' || $xivo_target_version == 'rc' || $xivo_target_version == 'five' ]]; then
```
The "[[" is a bashism and the whole test could be replaced by an sh-compatible test:

`[ $xivo_target_version = 'dev' -o $xivo_target_version = 'rc' -o $xivo_target_version = 'five' ]`

elessard1 commented on xivo_install_current.sh:
```diff
@@ -4,31 +4,49 @@ mirror_xivo="http://mirror.xivo.io"
+    if [[ $xivo_target_version == 'dev' || $xivo_target_version == 'rc' || $xivo_target_version == 'five' ]]; then
+        debian_version='8'
+    else
+        if [[ "$xivo_target_version" > "15.19" ]]; then
```
Same thing here, the [[ test could be replaced by:

`[ "$xivo_target_version" \> "15.19" ]`

elessard1 commented:
I've added some inline comments about a few of your questions.

Before xivo-dist, i.e. in 14.17 or earlier, to install an archive version, you had to install a "xivo-fai-xx.xx" package. The name of the package changed a few time since the 1.2 version. In this case, I feel like what the script should do is to display an error and quit when someone is trying to install an archive older than 14.18. So 14.18 would be the "oldest" archive version someone could install via this script.

gesnaud commented:
Hi Etienne,

I fixed whole things you observed.

I had a question, installing xivo with this modified script did not 
generate XIVO_UUID and did not create /etc/profile.d/xivo_uuid.sh.

PCM tell me, on irc, to execute xivo-configure-uuid to fix this issue... 
Do you think taht our modified scritp could the root cause?


Cheers and many thanks,
Gregory

On 06/22/16 14:42, Etienne Lessard wrote:
>
> I've added some inline comments about a few of your questions.
>
> Before xivo-dist, i.e. in 14.17 or earlier, to install an archive 
> version, you had to install a "xivo-fai-xx.xx" package. The name of 
> the package changed a few time since the 1.2 version. In this case, I 
> feel like what the script should do is to display an error and quit 
> when someone is trying to install an archive older than 14.18. So 
> 14.18 would be the "oldest" archive version someone could install via 
> this script.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub 
> <https://github.com/gesnaud/xivo-migration-scripts/pull/1#issuecomment-227731019>, 
> or mute the thread 
> <https://github.com/notifications/unsubscribe/ALhgMAOJhDf2YCAywXMyEj2FDiy35x67ks5qOS3NgaJpZM4I2GkE>.
>

-- 
*Grégory ESNAUD*
/*Expert Télécoms*

/
310 Route de Fougères - Bâtiment Europolis
35000 Rennes
Tel: +33(0)2 30 21 00 88 <tel://+33230210088>| Mob: +33(0)6 43 47 55 90 
<tel://+33643475590>| Support: 0811859486 <tel://0811859486>

*www.avencall.com*
<http://www.avencall.com>

*Suivez nous* <https://www.facebook.com/pages/AVENCALL/180804431983700> 
<http://twitter.com/#%21/avencall> 
<http://www.linkedin.com/company/avencall>

elessard1 commented:
I'm curious about the problem you had around the XIVO_UUID generation. Could you tell me the steps you have done, to see if I'm able to reproduce it ? i.e. how did you call the script / which archive version have you installed ?

Otherwise, I should be able to merge your changes later today.

elessard1 commented:
Since this pull request is made against your own repository, I created a branch in the xivo-pbx/xivo-migration-scripts repository, that you can consult here: https://github.com/xivo-pbx/xivo-migration-scripts/compare/install-archive-version . I've taken the occasion to modify another thing or two (like renaming the script).

I've also written some doc (and updated how the section was presented): https://github.com/xivo-pbx/xivo-doc/compare/install-archive-version

The two branches should be merged tomorrow. This pull request can be closed.

Thanks.

sduthil commented:
This PR is merged in upstream master, you may close it. Thanks a lot!
